File: CarCareHub_/CarCareHub.Model/ZaposlenikIzvjestaj.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarCareHub.Model
{

public class ZaposlenikIzvjestaj
    {
        public int ZaposlenikId { get; set; }
        public string ImePrezime { get; set; }
        public decimal UkupanIznos { get; set; }
        public int BrojNarudzbi { get; set; }
        public decimal ProsjecnaVrijednost { get; set; }
        public List<ProizvodStatistika> NajpopularnijiProizvodi { get; set; }
        public string Autoservis { get; set; } // Added autoservice info
    }
}

File: CarCareHub_/UI_/flutter_mobile/lib/models/zaposlenikIzvjestaj.dart
import 'package:flutter_mobile/models/autoservis.dart';
import 'package:flutter_mobile/models/drzave.dart';
import 'package:flutter_mobile/models/klijent.dart';
import 'package:flutter_mobile/models/proizvodiStatistika.dart';
import 'package:flutter_mobile/models/zaposlenik.dart';
import 'package:flutter_stripe/flutter_stripe.dart';
import 'package:json_annotation/json_annotation.dart';

part 'zaposlenikIzvjestaj.g.dart';

@JsonSerializable()
class ZaposlenikIzvjestaj { 
  late final int zaposlenikId;
  late final String imePrezime;
  late final double ukupanIznos;
  late final int brojNarudzbi;
  late final double prosjecnaVrijednost;
  late final List<ProizvodStatistika> najpopularnijiProizvodi;
  late final String autoservis;

  ZaposlenikIzvjestaj({
    required this.zaposlenikId,
    required this.imePrezime,
    required this.ukupanIznos,
    required this.brojNarudzbi,
    required this.prosjecnaVrijednost,
    required this.najpopularnijiProizvodi,
    required this.autoservis,
  });


  factory ZaposlenikIzvjestaj.fromJson(Map<String, dynamic> json) => _$ZaposlenikIzvjestajFromJson(json);

  Map<String, dynamic> toJson() => _$ZaposlenikIzvjestajToJson(this);
}